modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -130,6 +130,53 @@ export class {{classname}} {
 {{/operation}}
 {{/useHttpClient}}
 
+    private addToHttpParams(
+        httpParams: {{#useHttpClient}}HttpParams{{/useHttpClient}}{{^useHttpClient}}URLSearchParams{{/useHttpClient}},
+        value: any, key?: string): {{#useHttpClient}}HttpParams{{/useHttpClient}}{{^useHttpClient}}URLSearchParams{{/useHttpClient}}
+    {
+        if (typeof value === "object") {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParams1(httpParams, value);
+        } else {
+            {{#useHttpClient}}httpParams = {{/useHttpClient}}this.addToHttpParams1(httpParams, value, key);
+        }
+        return httpParams;
+    }
+
+    private addToHttpParams1(
```

Comment:
How could we improve the naming of this method?

Reply:
Well, this is our (project) standard naming for this kind of recursion. What about `addToHttpParamsRec`?